packages/editor-ui/src/composables/useRunWorkflow.ts

```diff
@@ -0,0 +1,377 @@
+import type {
+	IExecutionPushResponse,
+	IExecutionResponse,
+	IStartRunData,
+	IWorkflowDb,
+} from '@/Interface';
+import type {
+	IDataObject,
+	IRunData,
+	IRunExecutionData,
+	ITaskData,
+	IPinData,
+	IWorkflowBase,
+	Workflow,
+} from 'n8n-workflow';
+import {
+	NodeHelpers,
+	NodeConnectionType,
+	TelemetryHelpers,
+	FORM_TRIGGER_PATH_IDENTIFIER,
+} from 'n8n-workflow';
+
+import { useToast } from '@/composables/useToast';
+import { useNodeHelpers } from '@/composables/useNodeHelpers';
+
+import { FORM_TRIGGER_NODE_TYPE, WAIT_NODE_TYPE } from '@/constants';
+import { useTitleChange } from '@/composables/useTitleChange';
+import { useRootStore } from '@/stores/n8nRoot.store';
+import { useUIStore } from '@/stores/ui.store';
+import { useWorkflowsStore } from '@/stores/workflows.store';
+import { openPopUpWindow } from '@/utils/executionUtils';
+import { useExternalHooks } from '@/composables/useExternalHooks';
+import { useWorkflowHelpers } from '@/composables/useWorkflowHelpers';
+import { useRouter } from 'vue-router';
+import { isEmpty } from '@/utils/typesUtils';
+import { useI18n } from '@/composables/useI18n';
+import { useTelemetry } from '@/composables/useTelemetry';
+
+export function useRunWorkflow() {
+	const nodeHelpers = useNodeHelpers();
+	const router = useRouter();
```

Comment:
We can only use `useRouter` composable if called within the component instance.  So, if this composable would be imported from another composable, the router would be undefined which might be obvious. This is the reason we inject it into `useWorkflowHelpers`. Perhaps we should do the same hare, or at least thrown an error immediately on set-up if router is undefined?